common/common-client/src/main/java/org/ow2/proactive/authentication/IAMLoginModule.java

```diff
@@ -0,0 +1,393 @@
+ *
+ *
+ * based on Apereo CAS framework
+ *
+ *
+ * @author The ActiveEon Team
+ * @since ProActive Scheduling 8.2
+ */
+public class IAMLoginModule implements LoginModule {
+
+    /** Logger instance */
+    private static final Logger LOG = Logger.getLogger(IAMLoginModule.class.getName());
+
+    private static final String IAM_URL_KEY = "iamServerUrlPrefix";
+
+    /** JAAS authentication subject */
+    private Subject subject;
+
+    /** JAAS callback handler */
+    private CallbackHandler callbackHandler;
+
+    /** IAM Server URL */
+    private String iamServerUrlPrefix;
+
+    /** IAM REST request for tickets */
+    private String iamTicketRequest;
+
+    /** IAM Ticket/Token validator */
+    private String ticketValidatorClass;
+
+    /** IAM Ticket/Token validator extracted from ticketValidatorClass*/
+    private TicketValidator ticketValidator;
+
+    /** IAM client service */
+    private String service;
+
+    /** IAM assertion */
+    private Assertion assertion;
+
+    /** Name of the attribute in the CAS assertion that should be used for the username */
+    private String userAttributeName;
+
+    /** character used to separate many roles assigned to the user */
+    private String roleSeparator;
+
+    /** Name of the attribute in the CAS assertion that should be used for user role data */
+    private String roleAttributeName;
+
+    /** IAM response header used to get the SSO ticket */
+    private String ssoTicketHeader;
+
+    /** Boolean mentioning whether the JWT is signed */
+    private boolean jsonWebTokenSigned;
+
+    /** Boolean mentioning whether the JWT encrypted */
+    private boolean jsonWebTokenEncrypted;
+
+    /** JWT signature key */
+    private String jsonWebTokenSignatureKey;
+
+    /** JWT encryption key */
+    private String jsonWebTokenEncryptionKey;
+
+    /** login status */
+    private boolean succeeded = false;
+
+    /**
+     * Initialize this <code>IAMLoginModule</code>.
+     *
+     * <p>
+     *
+     * @param subject
+     *            the <code>Subject</code> not to be authenticated.
+     *            <p>
+     *
+     * @param callbackHandler
+     *            a <code>CallbackHandler</code> to get the credentials of the
+     *            user, must work with <code>NoCallback</code> callbacks.
+     *            <p>
+     * @param sharedState state shared with other configured LoginModules. <p>
+     *
+     * @param options options specified in the login
+     *			<code>Configuration</code> for this particular
+     *			<code>IAMLoginModule</code>.
+     */
+    @Override
+    public void initialize(Subject subject, CallbackHandler callbackHandler, Map<String, ?> sharedState,
+            Map<String, ?> options) {
+        this.subject = subject;
+        this.callbackHandler = callbackHandler;
+        this.assertion = null;
+        this.ticketValidatorClass = null;
+
+        for (final Map.Entry entry : options.entrySet()) {
```

Comment:
I do not see the map entries being validated against `null` and against empty strings. Are you sure you always have a usable key and value in the following?

Reply:
Done